locals.tf

```diff
@@ -0,0 +1,5 @@
+locals {
+  model_name       = var.model_name != "" ? var.model_name : "${var.resource_naming_prefix}-model"
```

Comment:
Constructing each of these names as locals means we can make the associated naming variables optional.

Reply:
It means that if they don't provide the model name it defaults to `${var.resource_naming_prefix}-model`. But I agree that it's all confusing atm... I'm going to refactor the naming so we just use `resource_naming_prefix` for everything

Reply:
Done